src/utils/event_manager.py

```diff
@@ -78,27 +78,23 @@ def _process_future(self, future, service):
                 item, timestamp = result
             else:
                 item, timestamp = result, datetime.now()
+            if item and not hasattr(item, "_id"):
+                store_or_update_item(item)
```

Comment:
Isnt this already done in the _run_thread_with_db_item method?

Reply:
they get processed first here, in which case they'll have to have an id to go further?